spec/models/ems_refresh/parsers/kubernetes_spec.rb

```diff
@@ -62,4 +62,24 @@
       end
     end
   end
+
+  describe "parse_container_image" do
+    shared_image = "shared/image"
+    shared_ref = "shared:ref"
+    unique_ref = "unique:ref"
+
+    it "returns unique object for same image but different ref/id" do
+      first_obj  = parser.parse_container_image(shared_image, shared_ref)
+      second_obj = parser.parse_container_image(shared_image, unique_ref)
+
+      first_obj.should_not == second_obj
+    end
+
+    it "returns same object for same image and ref/id" do
+      first_obj  = parser.parse_container_image(shared_image, shared_ref)
+      second_obj = parser.parse_container_image(shared_image, shared_ref)
+
+      first_obj.should == second_obj
```

Comment:
@EdDev `==` is especially wrong here as it's supposed to be `true` only for the same object. In fact:

    {:name => 'shared/image', :id => 'shared:ref'} == {:name => 'shared/image', :id => 'shared:ref'}

returns `true`, but it should fail this test (two different objects).